last_5_so_po/models/product_template.py

```diff
@@ -0,0 +1,46 @@
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    # Added fields #T00447
+    last_5_po_ids = fields.One2many(
+        comodel_name="purchase.order",
+        inverse_name="product_id",
+        compute="_compute_last_5_po",
+        string="Purchase orders",
+    )
+
+    last_5_so_ids = fields.One2many(
+        comodel_name="sale.order",
+        inverse_name="product_id",
+        compute="_compute_last_5_so",
+        string="Sale orders",
+    )
+
+    def _compute_last_5_po(self):
+        """compute method to show last 5 po of a product #T00447"""
+        for product in self:
+            purchase_orders = self.env["purchase.order"].search(
+                [
+                    ("order_line.product_id.id", "in", product.product_variant_ids.ids),
+                    ("state", "=", "purchase"),
+                ],
+                order="date_approve desc",
+                limit=5,
+            )
+            self.write({"last_5_po_ids": purchase_orders})
+
+    def _compute_last_5_so(self):
+        """Compute method to show last 5 SO of a product #T00447"""
+        for product in self:
+            sale_orders = self.env["sale.order"].search(
+                [
+                    ("order_line.product_id.id", "in", product.product_variant_ids.ids),
+                    ("state", "=", "sale"),
+                ],
+                order="date_order desc",
+                limit=5,
+            )
```

Comment:
`sale_orders = self.env["sale.order"].search([`
   ` ("order_line.product_id", "in", product.product_variant_ids.ids),`
    `("state", "=", "sale"),`
`], order="date_order desc", limit=5)`


"order_line.product_id" is used directly to search for the product IDs associated with the order lines. The use of "id" in the field name is unnecessary as product_id is sufficient.